src/components/organisms/NewProfileModal/components/header/ProfileModalAvatar/ProfileModalAvatar.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useRef, useState } from "react";
+import { useForm } from "react-hook-form";
+import classNames from "classnames";
+
+import { ACCEPTED_IMAGE_TYPES } from "settings";
+
+import { User } from "types/User";
+import { ContainerClassName } from "types/utility";
+
+import { WithId } from "utils/id";
+import { userProfileModalFormProp as formProp } from "utils/propName";
+
+import { useIsCurrentUser } from "hooks/useIsCurrentUser";
+import { useShowHide } from "hooks/useShowHide";
+import { useUploadProfilePictureHandler } from "hooks/useUploadProfilePictureHandler";
+import { useUser } from "hooks/useUser";
+
+import { UserAvatar } from "components/atoms/UserAvatar";
+
+import "firebase/storage";
+
+import "./ProfileModalAvatar.scss";
+
+interface Props extends ContainerClassName {
+  user: WithId<User>;
+  editMode?: boolean;
+  setPictureUrl?: (url: string) => void;
+  pictureUrl?: string;
+  register?: ReturnType<typeof useForm>["register"];
+}
+
+export const ProfileModalAvatar: React.FC<Props> = ({
+  editMode,
+  user,
+  register,
+  pictureUrl,
+  setPictureUrl,
+  containerClassName,
+}: Props) => {
+  const isCurrentUser = useIsCurrentUser(user);
+
+  const uploadRef = useRef<HTMLInputElement>(null);
+  const [error, setError] = useState("");
+
+  const {
+    isShown: uploading,
```

Comment:
```suggestion
    isShown: isUploading,
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1934/commits/a759004a4604dba81c85314f92fb5042ddff9306